pkg/liqo-controller-manager/namespaceOffloading-controller/clusterselector_management.go

```diff
@@ -0,0 +1,85 @@
+package namespaceoffloadingctrl
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	k8shelper "k8s.io/component-helpers/scheduling/corev1"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+)
+
+func (r *NamespaceOffloadingReconciler) selectCompliantVirtualNodes(noff *offv1alpha1.NamespaceOffloading,
+	clusterIDMap map[string]*mapsv1alpha1.NamespaceMap) error {
+	virtualNodes := &corev1.NodeList{}
+	if err := r.List(context.TODO(), virtualNodes,
+		client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode}); err != nil {
+		klog.Error(err, " --> Unable to List all virtual nodes")
+		return err
+	}
+
+	// If here there are no virtual nodes is an error because it means that in the cluster there are NamespaceMap
+	// but not their associated virtual nodes
+	if len(virtualNodes.Items) == 0 {
```

Comment:
I suppose every mismatch between the number of virtual nodes and namespace maps should be treated as an error

Reply:
you are right, ty